### PR TITLE
INTMDB-157: Add deprecation notice

### PR DESCRIPTION
**Pull request by themantissa**
## Description

Add deprecation notice to both programmatic key whitelist and project ip whitelist, replaced with access list.

Link to any related issue(s): 

## Type of change:

Adds simple deprecation notice in code

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/whitelist_api_keys.go [hunk 1]
```diff
@@ -6,6 +6,13 @@ import (
 	"net/http"
 )
 
+/* IMPORTANT: DEPRECATION NOTICE
+Access List Replaces Whitelist
+Atlas now refers to programmatic API key whitelists as access lists.
+Atlas has deprecated the whitelist method and will disable it in June 2021.
+Please update any dependent work to use accesslist_api_keys.go
+*/
```
Comment (jamesbroadhead) [inline mongodbatlas/whitelist_api_keys.go:14]: I think if we follow this format https://github.com/golang/go/wiki/Deprecated then tools will pick up the fact that this is a deprecation announcement automatically

**Review by jamesbroadhead — DISMISSED** [review]

File: mongodbatlas/whitelist_api_keys.go (hunk 1 above)
Comment (themantissa) [reply mongodbatlas/whitelist_api_keys.go:14]: Thank you so much @jamesbroadhead, did not know that!   Added commit to fit the spec.

File: mongodbatlas/project_ip_whitelist.go [hunk 2]
```diff
@@ -7,6 +7,13 @@ import (
 	"net/url"
 )
 
+/* IMPORTANT: DEPRECATION NOTICE
+Deprecated: Access List Replaces Whitelist
```
Comment (gssbzn) [inline mongodbatlas/project_ip_whitelist.go:11]: [super-nit] can we move this below line 21? james cover how to do this in go but right now making this a global comment makes the whole package (`mongodbatlas`) deprecated, you want to flag that only the `ProjectIPWhitelistService` is the one being deprecated 

Same with `WhitelistAPIKeysService` and I think we can skip commenting on the test file

**Review by gssbzn — CHANGES_REQUESTED** [review]

File: mongodbatlas/project_ip_whitelist.go (hunk 2 above)
Comment (themantissa) [reply mongodbatlas/project_ip_whitelist.go]: Removed from tests and moved lower in latest commit.

**Review by gssbzn — APPROVED** [review]
Sorry, pointing to the line number is too complex so I pushed something here to avoid going in circles, also this kind of works
Screenshot 2021-01-11 at 18 45 17

LGTM

**Comment by themantissa** [comment]
@gssbzn sometimes it's easier to do than explain - thank you :)   @jamesbroadhead  if you are good I'll merge this tomorrow so we can release (or you can, either works)

**Comment by jamesbroadhead** [comment]
oh yep - thanks for testing it out @gssbzn !

**Review by jamesbroadhead — APPROVED** [review]

**Pull request merged**